### PR TITLE
drm GL backend performance fix

### PR DESCRIPTION
**Pull request by anholt**
A Raspberry Pi user submitted a bug report to me about performance of the open source graphics driver with RetroArch (https://github.com/anholt/linux/issues/57) with max swapchain of 2, but the first thing I found looks like a bug in RetroArch's drm_ctx.c.

This is very lightly tested -- just watching the vblank missed complaints during the splash screen of an NES game I had laying around, using a nightly nestopia_libretro.so.

The real problem, I think, is that we immediately block on the flip completion after swap, to make the back buffer go idle.  However, I think it will be a long time until the next gl_frame(), causing us to miss the vblank.  I'm assuming nobody's going to be drawing to the default framebuffer until gl_frame(), so we could just block on completion at that point.  Is there a reasonable way to hook that up?

### CODE REVIEW

**Comment by anholt** [comment]
Force-pushed.  I realized that I'd made an order-of-magnitude mistake in reading the timing data.  Still probably worth it.

**Comment by Brunnis** [comment]
I'd hold off on merging this as I think it may not have the intended effect. See discussion in https://github.com/anholt/linux/issues/57

**Comment by twinaphex** [comment]
@Themaister Would you have anything to add to the discussion being held here?

https://github.com/anholt/linux/issues/57

File: gfx/drivers_context/drm_ctx.c [hunk 1]
```diff
@@ -215,9 +215,10 @@ static bool gfx_ctx_drm_queue_flip(void)
    fb                = (struct drm_fb*)drm_fb_get_from_bo(g_next_bo);
 
    if (drmModePageFlip(g_drm_fd, g_crtc_id, fb->fb_id,
-         DRM_MODE_PAGE_FLIP_EVENT, &waiting_for_flip) == 0)
+                       DRM_MODE_PAGE_FLIP_EVENT | DRM_MODE_PAGE_FLIP_ASYNC,
```
Comment (Themaister) [inline gfx/drivers_context/drm_ctx.c]: This is the only change I see. Can you explain what this does, and what it accomplishes? I haven't seen this flag before.

**Comment by anholt** [comment]
I was in fact mistaken about the flag (we used to have "async" meaning two things, and the thing I wanted got renamed to "nonblock" recently to clarify it.)

The blocking on the last flip immediately after flipping is still a problem, but let's close this for now.

**Pull request closed without merging**